homeassistant/components/lektrico/sensor.py

```diff
@@ -0,0 +1,296 @@
+    LektricoSensorEntityDescription(
+        key="power_l1",
+        translation_key="power_l1",
+        device_class=SensorDeviceClass.POWER,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfPower.KILO_WATT,
+        value_fn=lambda data: float(data.power_l1) / 1000,
+    ),
+    LektricoSensorEntityDescription(
+        key="power_l2",
+        translation_key="power_l2",
+        device_class=SensorDeviceClass.POWER,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfPower.KILO_WATT,
+        value_fn=lambda data: float(data.power_l2) / 1000,
+    ),
+    LektricoSensorEntityDescription(
+        key="power_l3",
+        translation_key="power_l3",
+        device_class=SensorDeviceClass.POWER,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfPower.KILO_WATT,
+        value_fn=lambda data: float(data.power_l3) / 1000,
+    ),
+    LektricoSensorEntityDescription(
+        key="pf_l1",
+        translation_key="pf_l1",
+        device_class=SensorDeviceClass.POWER_FACTOR,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=PERCENTAGE,
+        value_fn=lambda data: float(data.power_factor_l1) * 100,
+    ),
+    LektricoSensorEntityDescription(
+        key="pf_l2",
+        translation_key="pf_l2",
+        device_class=SensorDeviceClass.POWER_FACTOR,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=PERCENTAGE,
+        value_fn=lambda data: float(data.power_factor_l2) * 100,
+    ),
+    LektricoSensorEntityDescription(
+        key="pf_l3",
+        translation_key="pf_l3",
+        device_class=SensorDeviceClass.POWER_FACTOR,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=PERCENTAGE,
+        value_fn=lambda data: float(data.power_factor_l3) * 100,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Lektrico charger based on a config entry."""
+    coordinator: LektricoDeviceDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    _sensors_to_be_used: tuple[LektricoSensorEntityDescription, ...]
+    if coordinator.device_type == Device.TYPE_1P7K:
+        _sensors_to_be_used = SENSORS_FOR_CHARGERS + SENSORS_FOR_1_PHASE
+    elif coordinator.device_type == Device.TYPE_3P22K:
+        _sensors_to_be_used = SENSORS_FOR_CHARGERS + SENSORS_FOR_3_PHASE
+    elif coordinator.device_type == Device.TYPE_EM:
+        _sensors_to_be_used = (
+            SENSORS_FOR_LB_DEVICES + SENSORS_FOR_1_PHASE + SENSORS_FOR_LB_1_PHASE
+        )
+    elif coordinator.device_type == Device.TYPE_3EM:
+        _sensors_to_be_used = (
+            SENSORS_FOR_LB_DEVICES + SENSORS_FOR_3_PHASE + SENSORS_FOR_LB_3_PHASE
+        )
+    else:
+        raise PlatformNotReady
```

Comment:
Why do we raise this when a new device is added?

Reply:
When a new device is added and its type is not supported, we need to raise an exception. 
I changed it to IntegrationError.

Reply:
I think we should just do nothing instead of raising errors